lib/s3_meta_sync/syncer.rb

```diff
@@ -244,21 +263,26 @@ def download_content(path)
           "https://s3#{"-#{region}" if region}.amazonaws.com/#{@bucket}/#{path}"
         end
       options = (@config[:ssl_none] ? {:ssl_verify_mode => OpenSSL::SSL::VERIFY_NONE} : {})
-      open(url, options)
-    rescue OpenURI::HTTPError
-      retries ||= 0
-      retries += 1
-      if retries >= 3
+      with_retries(path: path, url: url) { open(url, options) }
+    end
+
+    def with_retries(path:, url:)
```

Comment:
this is not a generic retry logic so maybe pick a more specific name `retry_downloads` or so